Makefile

```diff
@@ -129,6 +129,11 @@ test-mk-do: manifests generate fmt vet envtest generate-deploy ## Run tests agai
 test-mk-do-fast: manifests generate fmt vet envtest generate-deploy ## Run tests against minikube with deployed operator(do) and exclude slow
 	DEPLOY_OPERATOR=true USE_EXISTING_CLUSTER=true ENABLE_WEBHOOKS=false go test  ./controllers/... -test.timeout=40m -ginkgo.slow-spec-threshold=30s -ginkgo.fail-fast -ginkgo.v -ginkgo.label-filter="do && !slow"
 
+## It requires a registry to push the image for testing
+## example: make OPERATOR_IMAGE_REPO=quay.io/hgao/operator OPERATOR_VERSION=1.0 test-mk-doh
+test-mk-doh: manifests generate fmt vet envtest generate-deploy docker-build docker-push ## Run tests against minikube with deployed operator image built from repo head(doh)
+	DEPLOY_OPERATOR=true USE_EXISTING_CLUSTER=true ENABLE_WEBHOOKS=false go test  ./controllers/... -test.timeout=40m -ginkgo.slow-spec-threshold=30s -ginkgo.fail-fast -ginkgo.v -ginkgo.label-filter="doh"
+
```

Comment:
if we change the test restarting the operator we don't need to deploy it, i.e.
```
// Restart the controller manager
shutdownControllerManager()
createControllerManagerForSuite()
```

Reply:
I need to test a real operator deployed into k8s/openshift. The shutdownControllerManager/createControllerManager cannot serve the purpose, it runs a local controller manager which is not exactly build from operator's main.go.

Reply:
I see, why do wee need to add `test-mk-doh` could we reuse `test-mk-do`?

Reply:
test-mk-doh will build the operator from your local repo, push it to a repo and deploy the operator. test-mk-do just use the operator latest tag on artemiscloud repo. In my case I nee to build the operator to include my fix and test it.
so I added 'test-mk-doh' for that matter, and also it will take longer time to build and test.

Reply:
that test-mk-do probably should do that. 
I am using it via:
```
make OPERATOR_IMAGE_REPO=localhost/am OPERATOR_VERSION=v1.0 build
podman-remote build -t am:v1.0 .
make OPERATOR_IMAGE_REPO=localhost/am OPERATOR_VERSION=v1.0 test-mk-do
```

in this way, the deployed yamls would reference the am:v1.0 image and not the latest.
the podman-remote build gets the image loaded into my local podman registry on the minikube vm

Reply:
does it work on openshift?

Reply:
with a different registry or with openshift access to the local podman registry it should.
I guess the build and push steps will be necessary to automate when it is not in code.

Reply:
ok, so the target seems redundant. I'll remove it.